File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String str) {
        Stack<Character> stack = new Stack<Character>();
		for(char c : str.toCharArray()) {
			if(c == '(') {
				stack.push(c);
			} else {
				if(stack.isEmpty()) {
					return false;
				} else {
					stack.pop();
				}
			}
		}
		if(!stack.isEmpty()) {
			return false;
		} else {
			return true;
		}
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
const solution = (s) => {
    const stack = [];
    
    for(const char of s) {
        if(char === '(') {
            stack.push(char);
        } else {
            if(!stack.length) {
                return false;
            } else {
                stack.pop();
            }
        }
    }
    
    return stack.length ? false : true;
}